openstack/04-TRAN/opencoap.c

```diff
@@ -331,7 +331,7 @@ void opencoap_sendDone(OpenQueueEntry_t* msg, owerror_t error) {
 
 \post After this function returns, the msg contains 
 */
-void opencoap_writeLinks(OpenQueueEntry_t* msg) {
+void opencoap_writeLinks(OpenQueueEntry_t* msg, uint8_t componentID) {

```

Comment:
The idea is exactly to pass the componentID to opencoap_writeLinks for filtering resources. Let's take an example: a resource has got N subresources; N+1 resources will be registered into opencoap with the same componentID, with the root resource in the form /path0, and the other N in the form /path0/path1; a GET to the root resource calls opencoap_writeLinks passing the componentID; opencoap_writeLinks includes only the subresources into the ack packet (if they are discoverable).
If the componentID is COMPONENT_CWELLKNOWN, opencoap will select all resources in the form /path0 if they are discoverable.